### PR TITLE
bump cloudfoundry-client and add check_aws_conn to migrate-secrets

### PR DESCRIPTION
**Pull request by gabelton**
https://uktrade.atlassian.net/browse/DBTP-288

- bump cloudfoundry-client to fix 401 error when initialising client
- make `project-profile` a required flag for migrate-secrets command and validate with `check_aws_conn`

### CODE REVIEW

**Review by ejayesh — APPROVED** [review]
lgtm

File: commands/bootstrap_cli.py [hunk 1]
```diff
@@ -190,6 +192,7 @@ def migrate_secrets(config_file, env, svc, overwrite, dry_run):
     AWS_PROFILE=myaccount copilot-bootstrap.py ...
     """
 
+    check_aws_conn(project_profile)
```
Comment (WillGibson) [inline commands/bootstrap_cli.py:195]: I don't think we have this change covered by our tests. E.g. I commented it out and all the tests still passed.

Which reminds me, I keep meaning to look at adding mutation tests to our stuff.

**Review by shehzadashiq — APPROVED** [review]
LGTM

File: commands/bootstrap_cli.py (hunk 1 above)
Comment (yusufsheiqh) [reply commands/bootstrap_cli.py:195]: Agree with Will. Would be good to get a test in for this, if possible.

**Review by lgarvey — APPROVED** [review]
Nice 🚀

File: commands/bootstrap_cli.py (hunk 1 above)
Comment (rossmiller) [reply commands/bootstrap_cli.py:195]: @gabelton please add tests as per above prior to merge.

Comment (gabelton) [reply commands/bootstrap_cli.py:195]: Do you want a specific test for `aws_conn`? I had loads of failing tests when I made the changes, which I had to fix by adding various credentials logic and mocks to the migrate_secrets tests

Comment (WillGibson) [reply commands/bootstrap_cli.py:195]: In the context of this pull request, I think a test to check that `make_config` exits when `check_aws_conn` exits will be enough to guard against us accidentally removing that behaviour.

Comment (gabelton) [reply commands/bootstrap_cli.py:195]: Added a test for `migrate_secrets` and started adding tests for utils, but will create a separate ticket to finish doing the latter

**Review by yusufsheiqh — APPROVED** [review]

**Pull request merged**